wasmi/examples/tictactoe.rs

```diff
@@ -133,59 +128,40 @@ mod tictactoe {
 
 struct Runtime<'a> {
     player: tictactoe::Player,
+    made_turn: bool,
     game: &'a mut tictactoe::Game,
 }
 
-const SET_FUNC_INDEX: usize = 0;
-const GET_FUNC_INDEX: usize = 1;
-
-impl<'a> Externals for Runtime<'a> {
-    fn invoke_index(
-        &mut self,
-        index: usize,
-        args: RuntimeArgs,
-    ) -> Result<Option<RuntimeValue>, Trap> {
-        match index {
-            SET_FUNC_INDEX => {
-                let idx: i32 = args.nth(0);
-                self.game.set(idx, self.player)?;
-                Ok(None)
-            }
-            GET_FUNC_INDEX => {
-                let idx: i32 = args.nth(0);
-                let val: i32 = tictactoe::Player::into_i32(self.game.get(idx)?);
-                Ok(Some(val.into()))
-            }
-            _ => panic!("unknown function index"),
+#[wasmi_derive::derive_externals]
+impl<'a> Runtime<'a> {
```

Comment:
is this example tested anywhere?

because otherwise there are no tests!

Reply:
Also, there will no longer be example for implementing host *without* procedural macro?

Reply:
Yes, except from tests.

I don't think it is worthwhile to keep implementing `Externals` manually here since this example doesn't do anything interesting and can be easily changed with the proc-macro.